jsonrpc/eth_endpoint.go

```diff
@@ -677,30 +677,30 @@ func (e *Eth) EstimateGas(arg *txnArgs, rawNum *BlockNumber) (interface{}, error
 				// Specifying the transaction failed, but not providing an error
 				// is an indication that a valid error occurred due to low gas,
 				// which will increase the lower bound for the search
-				return true, nil
+				return true, []byte(hex.EncodeToString(result.ReturnValue)), nil
 			}
 
 			if isEVMRevertError(result.Err) {
 				// The EVM reverted during execution, attempt to extract the
 				// error message and return it
-				return true, constructErrorFromRevert(result)
+				return true, []byte(hex.EncodeToString(result.ReturnValue)), constructErrorFromRevert(result)
 			}
 
-			return true, result.Err
+			return true, []byte(hex.EncodeToString(result.ReturnValue)), result.Err
 		}
 
-		return false, nil
+		return false, []byte(hex.EncodeToString(result.ReturnValue)), nil
```

Comment:
We should not return return value in this case, because it is not being propagated. WDYT?

```suggestion
		return false, nil, nil
```

Reply:
So there can only happen gas error and revert error? And in any other failed case we won't have value for ReturnValue?